schemas/com.snowplowanalytics.monitor/application_status/jsonschema/1-0-0

```diff
@@ -0,0 +1,133 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for an application status context; describes the status of the application entity attached as context to an application status report",
+  "self": {
+    "vendor": "com.snowplowanalytics.monitor",
+    "name": "application_status",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+
+  "type": "object",
+  "properties": {
+    "periodLengthInSec": {
+      "description": "The length of the period to which this status applies; in seconds",
+      "type": "integer"
+    },
+    "periodStart": {
+      "description": "The start of the period to which this status applies",
+      "type": ["string", "null"],
+      "format": "date-time"
+    },
+    "periodEnd": {
+      "description": "The end of the period to which this status applies",
+      "type": ["string", "null"],
+      "format": "date-time"
+    },
+    "inAppStats": {
```

Comment:
When we developed the schema we distinguished between data points that would be emitted from the app with data points that could be generated by monitoring tools outside the app looking at the app.

It looks like in this schema the data points that would be self-reported by the app are nested in the `inAppStats` object, which I think is a nice approach. I'm curious why we wouldn't then nest the fields that would be recorded outside the app in an equivalent object? 

To be clear - I don't have a view on whether grouping and nesting these two categories of fields is a good thing to do - but I think whatever approach we take with one group we should also take with the other.

What do you think?

Reply:
The main reason I bundled these in the `inAppStats` array is to support reporting stats by tracker version. We don't know in advance how many tracker versions will be in use, so we need a flexible structure that can hold an arbitrary number of them.

If we do not need to slice by tracker version, then there is not real need for the `inAppStats` object and each field can just be a top-level property.

I did not take the same approach for the stats that are 'outside the app' because there we cannot guarantee that we can slice by tracker version. This schema is meant to not be specific to loaders. So, in some upstream process events that failed to be processed might not contain information about the tracker.

I think the best approach now would be to unnest these 'internal' stats.

Reply:
Agree